server/datastore/mysql/activities.go

```diff
@@ -19,17 +21,29 @@ func (ds *Datastore) NewActivity(ctx context.Context, user *fleet.User, activity
 
 	var userID *uint
 	var userName *string
+	var userEmail *string
 	if user != nil {
```

Comment:
What happens if `user == nil`? Do we still log the activity? If not, we should return an error and not try to write to DB.

Reply:
Yes, if `user == nil` we still create an `Activity`. E.g. a failed login results in an `Activity`, but the `user ` is nil.